File: Node/ava-js/test/example_test.js
const test = require ('ava');
const AxeDevToolsWebdriverJS = require('@axe-devtools/webdriverjs');
const  WebDriver = require('selenium-webdriver');
const { By } = require('selenium-webdriver');
const Reporter = require('@axe-devtools/reporter').default;
let reporter = new Reporter('DevTools-reporter', './a11y-results/homepageFlows');



test.beforeEach(async t => {
    t.context.driver = new WebDriver.Builder().forBrowser('chrome').build();
	await t.context.driver.get('https://broken-workshop.dequelabs.com/');
});


test('homepage', async homepage => {
    const {driver} = homepage.context;
    let axeDevTools =  new AxeDevToolsWebdriverJS(driver);
    // Analyzing foe a11y issues
    const results = await axeDevTools.analyze();
    // Logging the result using the reporter
    reporter.logTestResult('homepage-laptop-flow', results);
    homepage.pass();
}) 

test('recipeCard', async recipeCard => {
    const {driver} = recipeCard.context;
   await driver.findElement(By.css('#main-content > div.Recipes > div:nth-child(1) > div.Recipes__card-foot > button')).click();
   let axeDevTools =  new AxeDevToolsWebdriverJS(driver);
   // Analyzing foe a11y issues
   const results2 = await axeDevTools.analyze();
   // Logging the result using the reporter
   reporter.logTestResult('homepage-altered-state', results2);
   recipeCard.pass();
}) 


test.afterEach('cleanup', async t => {
	await t.context.driver.close();
});


test.after('reporting', async t => {
    await reporter.buildHTML('./a11y-results/html/');
    await reporter.buildJUnitXML('./a11y-results/xml/');
    await reporter.buildCSV('./a11y-results/csv/');
}); 

File: Java/java-selenium/src/test/java/example/exampleTest.java
package example;

import java.io.File;
import java.io.IOException;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.edge.EdgeDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import com.deque.html.axecore.results.Results;
import com.deque.html.axedevtools.selenium.*;
import com.deque.html.axedevtools.selenium.reporter.*;

public class exampleTest {
  private static AxeReportingOptions _reportOptions = new AxeReportingOptions();
  private static WebDriver driver = null;
  private static AxeDriver axedriver = null;
  private static AxeSelenium axeSelenium = null;

@BeforeClass
public static void initiate_drivers() {
    _reportOptions = new AxeReportingOptions();
    WebDriverManager.edgedriver().setup();
    driver = new EdgeDriver();  
    axedriver = new AxeDriver(driver);
    axeSelenium = new AxeSelenium();
    AxeConfiguration.configure().testSuiteName("Test").outputDirectory("axe-json-reports/");
}


  @Test
  public void test_a11y_workshop_homepage() {
    driver.get("https://broken-workshop.dequelabs.com/");
    Results results1 =
        axeSelenium.logResults(_reportOptions.uiState("Homepage_scan")).run(axedriver);
  }

  @Test
  public void test_recipe_card_a11y_workshop_homepage() {
    driver.get("https://broken-workshop.dequelabs.com/");
    driver
        .findElement(
            By.cssSelector(
                "#main-content > div.Recipes > div:nth-child(1) > div.Recipes__card-foot > button"))
        .click();
    Results results2 =
        axeSelenium.logResults(_reportOptions.uiState("Recipe_card_scan")).run(axedriver);
  }

  @AfterClass
  public static void reporting() throws IOException {
    Runtime rt = Runtime.getRuntime();
    String reporter = new File("src/test/resources/reporter.exe").getAbsolutePath();
    String Logger = new File("axe-json-reports/").getAbsolutePath();
    String Destination = new File("a11y-results/").getAbsolutePath();
    String command_xml =
        reporter + " " + Logger + " --destination " + Destination + " --format xml";
    String command_html =
        reporter + " " + Logger + " --destination " + Destination + " --format html";
    String command_csv =
        reporter + " " + Logger + " --destination " + Destination + " --format csv";
    try {
      rt.exec(command_html);
      rt.exec(command_csv);
      rt.exec(command_xml);
    } catch (Exception e) {
      e.printStackTrace();
    }
    driver.quit();
  }
}
